### ISSUE TITLE
Кран 334

### ISSUE THREAD

**Issue by AlmP-RUS** [metrostroi-repo/MetrostroiAddon#488]
Неверно работает кран 334

Не сажается срывной клапан в 3-м положении крана, хотя должен

![image](https://user-images.githubusercontent.com/79363684/165990419-f4aa6b1e-ddfc-4848-bea7-6cd2a6348224.png)
*Выдержка из Инструкции по приемке ПС от 2000г. (И-013/2000ТЭ)

Еще, вот что написано про 3-е положение крана в книге Федорова 1957 года по вагонам типа Д
![image](https://user-images.githubusercontent.com/79363684/165992938-4961be04-ddf5-4559-b022-9322fd6d04d2.png)

В метрострое кран работает явно не так (я проверял на версии метростроя из Стима)

И-013/2000ТЭ:
https://docs.google.com/document/d/1WWkLyr99laTyoaKLMQmauaJcHS6tJuoR/edit?usp=sharing&ouid=100288801166594546838&rtpof=true&sd=true

Вагоны типа Д - Федоров, 1957:
https://drive.google.com/file/d/1s5RDbmAbGNks5VdtqddLFzvfNALdELm6/view?usp=sharing

**Comment by AlmP-RUS** [comment]
Так же, кран не должен в 1 положении прям так сразу заряжать ТМ до уровня НМ, вот выписка из той же книжки Федорова
![image](https://user-images.githubusercontent.com/79363684/165994112-d18fdd2b-8508-4aec-8719-8f0e0719bf54.png)

**Comment by alexandr-sizov** [comment]
@BlueSnooty

**Comment by BlueSnooty** [comment]
Оке, посмотрю

**Comment by BlueSnooty** [comment]
Не обещаю, что переделаю весь кран, но баг с перекрышей устраню

**Comment by AlmP-RUS** [comment]
> Не обещаю, что переделаю весь кран, но баг с перекрышей устраню

Сделай так, чтоб при перекрыше ТМ не подпитывалсь от НМ, будь то стоп-кран или автостоп. Остальное уже не так важно. Ещё правда вопрос, как будет себя вести стрелка на манометре в такой ситуации, это надо ещё поискать

**Comment by BlueSnooty** [comment]
На самом деле там ТМ подпитывалась от УР. Уже исправил, скоро сделаю пул реквест

**Comment by BlueSnooty** [comment]
Ха-ха, там всё сложнее оказалось. Придется повозиться несколько дней

**Comment by AlmP-RUS** [comment]
Еще, в конспектах по номерному написано, что срывной клапан садится при давлении в ТМ 1,6-1,8 атм. В метрострое вроде как это происходит только при полном выходе воздуха из ТМ. На номере и Ешнике автостоп одинаковый, так что это тоже надо бы реализовать

**Comment by BlueSnooty** [comment]
Это я уже давно сделал (правда только для номерных). Даже ветку уже соединили с dev. Хреново то, что я не рассчитывал на то, что кому-то еще будет интересен 334 кран. А данное issue выявило некоторые недостатки моих правок пневматики, которые теперь надо устранить. Упс...

> Еще, в конспектах по номерному написано, что срывной клапан садится при давлении в ТМ 1,6-1,8 атм. В метрострое вроде как это происходит только при полном выходе воздуха из ТМ. На номере и Ешнике автостоп одинаковый, так что это тоже надо бы реализовать

**Comment by BlueSnooty** [comment]
Так, ситуация сдвинулась с места. Кстати, во втором положении КМ 334 не может разрядить ТМ, если она перезаряжена выше нормы. Только 4-м положением. Имейте в виду, чтобы потом не посчитали это багом :)

**Comment by BlueSnooty** [comment]
Как будто готово. Но теперь, после месяца работы с 334 краном, 013 кажется немного убогим >_>

**Comment by AlmP-RUS** [comment]
@BlueSnooty Меня тут в билетах на класс заинтересовал ответ на вопрос про наш чудесный кран
![image](https://user-images.githubusercontent.com/79363684/172806786-034f2851-5d54-4379-a46c-8413379ba129.png)
Кран будет в 3 положении работать, как тут написано?
Как я понял, кран машиниста в 3 положении не может заряжать ТМ, а только разряжать.
И еще, как себя будет вести стрелка на манометре, если зарядить тм и поставить кран в 3 положение, а потом разрядить?

**Comment by BlueSnooty** [comment]
@AlmP-RUS 
Попробую записать видео и выложу сюда ссылку. Но не сегодня

**Comment by BlueSnooty** [comment]
Могу сказать, что я реализовал утечку воздуха через неплотность поршня УР. А вот с реализмом при оставленных открытыми кранах двойной тяги в хвостовой кабине вышло не очень

**Comment by BlueSnooty** [comment]
@AlmP-RUS 
Нет, как написано в билете — к сожалению, не работает

**Comment by AlmP-RUS** [comment]
> @AlmP-RUS 
> Нет, как написано в билете — к сожалению, не работает

А надо сделать так, чтобы работало)

**Comment by BlueSnooty** [comment]
@AlmP-RUS 
Кстати, я немного соврал. Работает, но темпы разрядки совсем не такие, как в реале. И звук шипения привязан не к ТМ, а к УР, поэтому сначала мне показалось, что не работает. Звук я могу переделать, а вот с темпами разрядки-зарядки всё гораздо сложнее, и их не хотелось бы снова менять. В общем, на днях запишу видеодемонстрацию

**Comment by BlueSnooty** [comment]
@AlmP-RUS 
https://youtube.com/playlist?list=PL_7cacOBgjuDINklUuk45R6nLMqNClPIW
Вот демки работы КМ 334 после моих правок

**Comment by AlmP-RUS** [comment]
> @AlmP-RUS https://youtube.com/playlist?list=PL_7cacOBgjuDINklUuk45R6nLMqNClPIW Вот демки работы КМ 334 после моих правок

Все отлично, только два момента интересуют

1. Последнее видео немного не ясно, как я понял ты совал срывной при 2м положении крана, потом поставил в 3-е (зачем?)

2. ТМ слишком быстро заряжается 1 положением крана, возможно это уже конечно я докапываюсь на ровном месте, но в книге по Дшке есть такой момент:
![image](https://user-images.githubusercontent.com/79363684/174134820-2fdc6dcc-e963-4968-bd3c-28e90adabff5.png)
Тут написано, что давление в ТМ может _даже_ дойти до давления в НМ, тобишь, как я понял, темп зарядки тм (уже в районе 5 атм) будет падать с увеличением давления в ТМ, . В одном из видео была перезарядка, и мне уж этот момент зарядки (с 5атм до 8) показался уж очень быстрым.

**Comment by BlueSnooty** [comment]
По поводу скорости перезарядки я ничего не могу сделать. Можно, конечно, накостылить, но это нежелательно, так как может привести к непредсказуемым ошибкам в каких-нибудь ситуациях

А по поводу 3 положения: как "зачем"? Ты же сам спрашивал, почему в перекрыше срывной клапан не садится, вот я и сделал, чтобы он садился, как и должно быть

**Comment by BlueSnooty** [comment]
> Неверно работает кран 334
> 
> Не сажается срывной клапан в 3-м положении крана, хотя должен

Вот как бы твой первый пост в этом issue

**Comment by AlmP-RUS** [comment]
> > Неверно работает кран 334
> > 
> > Не сажается срывной клапан в 3-м положении крана, хотя должен
> 
> Вот как бы твой первый пост в этом issue

Я имел ввиду видос, а именно демонстрацию работы крана во 2 положении при срыве автостопа, ведь при работе на линии кран должен стоять в 3 положении

**Comment by AlmP-RUS** [comment]
А так, всё окей, все баги и недоработки пофикшены

**Comment by AlmP-RUS** [comment]
> А по поводу 3 положения: как "зачем"? Ты же сам спрашивал, почему в перекрыше срывной клапан не садится, вот я и сделал, чтобы он садился, как и должно быть


Просто видос называется "Работа на 3 положении", а в нём показано 2-е

**Comment by BlueSnooty** [comment]
![Screenshot_440](https://user-images.githubusercontent.com/89611276/174171889-e0ea73b2-f65a-4784-afc1-b9b463b74ddb.png)
Наверное, мы смотрим разные видео. Я в упор не вижу, где ты высмотрел "Работу на 3 положении" -__-
И при работе на линии кран никогда не должен стоять в 3 положении...

**Comment by AlmP-RUS** [comment]
> ![Screenshot_440](https://user-images.githubusercontent.com/89611276/174171889-e0ea73b2-f65a-4784-afc1-b9b463b74ddb.png)
> Наверное, мы смотрим разные видео. Я в упор не вижу, где ты высмотрел "Работу на 3 положении" -__-
> И при работе на линии кран никогда не должен стоять в 3 положении...

Посадка срывного перекрышей

Перекрыша = 3п

**Comment by AlmP-RUS** [comment]
> И при работе на линии кран никогда не должен стоять в 3 положении

Ну тут в разных депо по разному: меня например всегда учили, что в 3м  положении кран держать обязательно (товарищ из Новогиреево), фили/Измайлово - то же

**Comment by AlmP-RUS** [comment]
Видосы так же на ютубе есть из кабин, там то же кран в 3 стоит при работе на линии

**Comment by AlmP-RUS** [comment]
@BlueSnooty Извиняюсь за необоснованный докоп, только что всё перепроверил, глянул разные МИ, всё верно, проверка срывного верная, про положение крана при работе на линии ничего не нашёл, походу 3-е положение крана на линии- самодеятельность машинистов, а не предписание из МИ -__-

**Comment by BlueSnooty** [comment]
Ok :)
